src/server/config.js

```diff
@@ -72,14 +80,23 @@ if (process.env.ANALYZED_TEXT_FIELD_SUFFIX) {
   config.analyzedTextFieldSuffix = process.env.ANALYZED_TEXT_FIELD_SUFFIX;
 }
 
-// only three options for tier access level: 'private' (default), 'regular', and 'libre'
-if (process.env.TIER_ACCESS_LEVEL) {
-  if (process.env.TIER_ACCESS_LEVEL !== 'private'
-  && process.env.TIER_ACCESS_LEVEL !== 'regular'
-  && process.env.TIER_ACCESS_LEVEL !== 'libre') {
-    throw new Error(`Invalid TIER_ACCESS_LEVEL "${process.env.TIER_ACCESS_LEVEL}"`);
+// Either all indices should have explicit index-scoped tiered-access values or
+// the manifest should have a site-wide TIER_ACCESS_LEVEL value.
+// This approach is backwards-compatible with commons configured for past versions of tiered-access.
+let allIndicesHaveTierAccessSettings = true;
+config.esConfig.indices.forEach((item) => {
```

Comment:
Small thoughts: would it be better to validate the `tier_access_level` here for each indices, rather than throwing an error from `perIndexTierAccessMiddleware` ?

In this case you can abstract this validation into a function and apply it to both overall config and per-index config